### PR TITLE
stack explainer

### PR DESCRIPTION
**Pull request by zramsay**
Explains the role of - and links to - most of our repositories, documentation, and tutorials, while commenting on design considerations for building a chain-based application. Designed to be an "orientation document" for doing marmot things. 

Feedback welcome.

### CODE REVIEW

File: tutorials/the-eris-stack.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+4) Write you app (Dapp). We call them services. A handful are already built for you. Some of services are used internally, for example, [IPFS](http://ipfs.io/), to provide a data lake for applications (the `eris files` command). The [mindy](https://github.com/eris-ltd/mindy) [testnet](pinkpenguin.interblock.io:46657) has been running for many months now and works fairly well for sub-domain name registry. The [toadserver](https://github.com/eris-ltd/toadserver) integrates each mindy, IPFS, and eris:db's name registry feature to serve as a download server. [This will soon](https://github.com/eris-ltd/eris-cli/issues/579) replace github as part of the `eris init` sequence. I recently built the [marmot checker](https://github.com/eris-ltd/marmot) which integrates the Google Cloud Vision API with the toadserver.  Services glue together other services, and can often be added to you application with just a few lines of code.
```
Comment (compleatang) [inline tutorials/the-eris-stack.md]: Eris services are built to support modern, containerized, microservices styles applications out of the box.

File: tutorials/the-eris-stack.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Blockchain Checklist
+
+1) It all starts with your keys. This is crypto, remember? Key management is an inherently complex process and is something that should be given serious thought. It's getting easier overall (e.g. keybase), but at the moment there remain caveats. A key needs to be generated and secured for each participant in your blockchain. Can't do much on a blockchain without a validator or participant who can't sign transactions. This is a fundamentally different approach for web services compared to login/password. If you're gonna be blockchaining, do try to really understand how keys work and are used. Our key signing daemon [eris:keys](https://docs.erisindustries.com/documentation/ekeys/), usually run in a docker container, provides the basic functionality for getting started with development. As with all the tools in our stack, this key daemon can be swapped out for any other compatible cryptographic key tool. In fact, we anticipate moving to PGP in the future. Note: this keys daemon is for *development only* and we do not yet offer an enterprise key management solution.
```
Comment (compleatang) [inline tutorials/the-eris-stack.md]: Add some bolding around swapped out and basically say, it **must be swapped** out for production use cases. However the common keys API can either be satisfied by your audited security system of choice or we will be looking to provide proxy based access to various security and key management systems over time.

File: tutorials/the-eris-stack.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+4) Write you app (Dapp). We call them services. A handful are already built for you. Some of services are used internally, for example, [IPFS](http://ipfs.io/), to provide a data lake for applications (the `eris files` command). The [mindy](https://github.com/eris-ltd/mindy) [testnet](pinkpenguin.interblock.io:46657) has been running for many months now and works fairly well for sub-domain name registry. The [toadserver](https://github.com/eris-ltd/toadserver) integrates each mindy, IPFS, and eris:db's name registry feature to serve as a download server. [This will soon](https://github.com/eris-ltd/eris-cli/issues/579) replace github as part of the `eris init` sequence. I recently built the [marmot checker](https://github.com/eris-ltd/marmot) which integrates the Google Cloud Vision API with the toadserver.  Services glue together other services, and can often be added to you application with just a few lines of code.
+
+5) Deploy to the cloud. This process uses [docker-machine](https://docs.erisindustries.com/tutorials//tool-specific/docker_machine/) and will soon be bundled up in the `eris remotes` command to simplify the process even further.
```
Comment (compleatang) [inline tutorials/the-eris-stack.md]: use the advanced/cloud-getting-started here....

File: tutorials/the-eris-stack.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
...ne has a key that 1) can sign (via the keys daemon) and 2) is included in the genesis file, then it'll join the network. Once >2/3 of known validators join, the chain will begin validating transactions (i.e., updating state, "mining", producing blocks etc). If the chain drops below 2/3 validators, it will halt. The status and health of a chain can be monitored easily through various means such as the `mintinfo` command from mint-client, the js library (port 1337), or pure http endpoints (port 46657). The chains command for eris-cli is designed to handle all operations for one or more chains. With your chain running, you can send transactions (`mintx send`) and deploy contracts using eris:pm. At the tool level, these would primarily be used by chain admins whereas developers for a user-facing application would likely be working at the javascript layer for sending transactions and deploying contracts. More info: [chain deploying](https://docs.erisindustries.com/tutorials/advanced/chaindeploying/); [contracts interacting](https://docs.erisindustries.com/tutorials/contractsinteracting/); and [collaborating on your application](https://docs.erisindustries.com/tutorials/servicesmaking/).
```
Comment (compleatang) [inline tutorials/the-eris-stack.md]: `master` node => `first` node

File: tutorials/the-eris-stack.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+Now you need an application. Before we get into some design considerations for an app, let's dissect the process of sending transactions and deploying contracts. At the end of the day, the latter is the former but includes compiled byte code as part of the transaction (see eris-abi in Figure 1). The specific requirements for sending a transaction are documented [here](https://github.com/eris-ltd/mint-client/blob/master/README.md#walkabout). Provided these are met, the mintx command will first craft a transaction and, if specified, sign and broadcast it to the designated chain. Again, since the key signing that transaction must be available to the daemon, docker simplifies this logitics challenge. So that's pretty easy but it's no bitcoin wallet in terms of simplicity.
```
Comment (compleatang) [inline tutorials/the-eris-stack.md]: Hmm. I think we should say there are three ways to send tx's: (raw RPC commands, eris-pm, and eris-db.js)

File: tutorials/the-eris-stack.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
...ration tool. Everything we've discussed so far can be done natively (i.e., without docker); this requires stitching together the various tools (see, for example, [eris by curl](https://docs.erisindustries.com/tutorials/tool-specific/eris_by_curl/). eris:cli makes it easyier to co-ordinate all the things you need to roll a smart-contract ready chain with an application. With that in mind, our goal now is to build and define a service that, when started, links up to the existing chain and any other services that are required for the application to run smoothly. Once your service is built (write a bunch of code, basically), all it needs is a service definition file (`eris services new`; see the specification [here](https://docs.erisindustries.com/documentation/eris-cli/latest/services_specification/) which simplifies the `docker run` process. This of course assumes you've written a [Dockerfile](https://docs.docker.com/engine/reference/builder/) and made a [docker image](https://docs.docker.com/engine/userguide/containers/dockerimages/). See the [toadserver service making tutorial](https://docs.erisindustries.com/tutorials/advanced/servicesmaking/) for more information on this process.
```
Comment (compleatang) [inline tutorials/the-eris-stack.md]: easyier => easier

File: tutorials/the-eris-stack.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+This post is intended to provide a high level overview and a detailed explanation of how the eris stack is, well, stacked. [Modularity is a strong focus](https://eng.erisindustries.com/tutorials/2015/09/05/docker-and-eris/), greatly facilitated by [docker](https://docker.com). I'll be focusing on how the tools fit together while painting a path through the tutorial series. The complicated task of orchestrating all the various requirements for a smart-contract-focused blockchain is exactly why we went all-in on docker despite various ongoing pain points. We like to think of [eris:cli](https://docs.erisindustries.com/documentation/eris/) as "docker for blockchains".
```
Comment (compleatang) [inline tutorials/the-eris-stack.md]: Think we should move diagram to after this paragraph.

File: tutorials/the-eris-stack.md (hunk 5 above)
Comment (zramsay) [inline tutorials/the-eris-stack.md]: good call. this prgh was confusing

**Pull request merged**